src/transformers/models/mask2former/modeling_mask2former.py

```diff
@@ -0,0 +1,3393 @@
+            layer = PredictionBlock(in_dim, out_dim, activation=activation)
+            self.layers.append(layer)
+            # Provide backwards compatibility from when the class inherited from nn.Sequential
+            # In nn.Sequential subclasses, the name given to the layer is its index in the sequence.
+            # In nn.Module subclasses they derived from the instance attribute they are assigned to e.g.
+            # self.my_layer_name = Layer()
+            # We can't give instance attributes integer names i.e. self.0 is not permitted and so need to register
+            # explicitly
+            self.add_module(str(i), layer)
+
+    def forward(self, input: torch.Tensor) -> torch.Tensor:
+        hidden_state = input
+        for layer in self.layers:
+            hidden_state = layer(hidden_state)
+        return hidden_state
+
+
+class Mask2FormerMaskPredictor(nn.Module):
+    def __init__(self, config: DetrConfig, mask_feature_size: torch.Tensor):
+        super().__init__()
+        self.hidden_size = config.hidden_size
+        self.num_heads = config.decoder_attention_heads
+
+        self.mask_embedder = Mask2formerMLPPredictionHead(self.hidden_size, self.hidden_size, mask_feature_size)
+
+    def forward(self, outputs: torch.Tensor,
+            pixel_embeddings: torch.Tensor,
+            attention_mask_target_size: int = None
+            ):
+        
+        outputs = outputs.transpose(0,1)
+        # get the masks
+        mask_embeddings = self.mask_embedder(outputs)
+        # sum up over the channels
+        outputs_mask = torch.einsum("bqc,   bchw -> bqhw", mask_embeddings, pixel_embeddings)
+
+        attention_mask = F.interpolate(outputs_mask, size=attention_mask_target_size, mode="bilinear", align_corners=False)
+        print("attention_mask1:",attention_mask.shape)
+        attention_mask = (attention_mask.sigmoid().flatten(2).unsqueeze(1).repeat(1, self.num_heads, 1,1).flatten(0,1) < 0.5).bool()
+        print("attention_mask2:",attention_mask.shape)
+        attention_mask = attention_mask.detach()
+        print("attention_mask3:",attention_mask.shape)
+    
+        return outputs_mask, attention_mask
+
+
+class Mask2FormerHungarianMatcher(nn.Module):
+    """This class computes an assignment between the labels and the predictions of the network.
+
+    For efficiency reasons, the labels don't include the no_object. Because of this, in general, there are more
+    predictions than labels. In this case, we do a 1-to-1 matching of the best predictions, while the others are
+    un-matched (and thus treated as non-objects).
+    """
+
+    def __init__(self, num_points: int, cost_class: float = 1.0, cost_mask: float = 1.0, cost_dice: float = 1.0):
+        """Creates the matcher
+
+        Params:
+            cost_class (float, *optional*, defaults to 1.0):
+                This is the relative weight of the classification error in the matching cost.
```

Comment:
Looks great, I just a have a few suggestions to make the docstrings a bit more concise.
```suggestion
                Relative weight of the classification error in the matching cost.
```